monkey/monkey_island/cc/ui/src/components/pages/MapPage.js

```diff
@@ -138,16 +150,46 @@ class MapPageComponent extends AuthComponent {
     );
   }
 
+  handleScroll(e) {
+    let element = e.target;
+
+    var telemetryStyle = window.getComputedStyle(element)
+    var telemetryLineHeight = parseInt((telemetryStyle.lineHeight).replace('px', ''))
+
+    this.setState({
```

Comment:
You can move this entire `setState` inside the `setState` present in `if (element.scrollTop < this.scrollTop) ` if-else loop.